docs/reference_guides/core/costing/costing_framework.rst

```diff
@@ -0,0 +1,70 @@
+
+* `UnitModel` stands for a unit model block - users should supply an IDAES or custom unit model
+* `ParameterBlock` stands for a property package parameter block - users should supply an IDAES or custom property package
+* `UnitModelCostingBlock` is the required class to define the costing sub-block
+* `FlowsheetCostingBlock` is the required class to define the flowsheet-level costing level; when costing individual unit models with a custom costing package, inheritance allows the costing class to stand in for this block. Below, `CustomPackageCosting` inherits all flowsheet-level costing block properties and may be invoked instead.
+* `FlowsheetCostingBlock` is the required costing data class, which likewise may be defined by invoking a `CustomPackageCostingData` via inheritance.
+* `unit_costing_method` is a specific costing method that IDAES will pass to define costing variables and constraints. This is completely up to the user and grants freedom and flexibility to integrate custom costing for specific unit models.
+* `CustomPackageOption1` and `CustomPackageOption2` are example costing method arguments, which pass an `option` to the costing block.
+
+
+.. code:: python
+
+    from pyomo.environ import ConcreteModel
+    from idaes.core import FlowsheetBlock
+    from custom_properties import ParameterBlock
+    from idaes.models.unit_models import UnitModel
+
+    from idaes.core import UnitModelCostingBlock
+    from idaes.models.costing.CustomPackage import CustomPackageCosting, CustomPackageCostingData
+    from idaes.models.costing.CustomPackage import CustomPackageOption1, CustomPackageOption2
+    
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    
+    m.fs.properties = ParameterBlock()  # this is a relevant thermodynamic property package
+    
+    m.fs.unit = UnitModel()  # this is a unit model supported by the custom costing package
+    # set inputs
+    # ...
+    # adding flow, temperature, pressure, other state and unit specifications
+    
+    m.fs.costing = CustomPackageCosting()
+    m.fs.unit.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.costing,
+            "costing_method": CustomPackageCostingData.unit_costing_method,
+            "costing_method_arguments": {
+                    "option_1": CustomPackageOption1.option,
+                    "option_2": CustomPackageOption2.option
+            }
+        }
+    )
+
+Units of Measurement
+--------------------
+
+It is important to highlight that the costing method interrogates the property package to determine the units of measurement of this model. For example, if a costing package defines area in :math:`m^2`, while the cost correlations for heat exchangers require units to be in :math:`ft^2`, the costing method will convert the units to :math:`ft^2`. The use of Pyomo-unit conversion tools is under development.
```

Comment:
A few things here:

1. Pyomo units tool is not under development, it is finished.
2. The costing method does not interrogate the property package for units - the units are part of the variables and expressions themselves. Just say that units are converted as necessary (and maybe point to the Pyomo docs for units).